sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/OpenAIClientNonAzureImpl.java

```diff
@@ -0,0 +1,721 @@
+import com.azure.core.annotation.HeaderParam;
+import com.azure.core.annotation.Host;
+import com.azure.core.annotation.HostParam;
+import com.azure.core.annotation.Post;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceInterface;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.annotation.UnexpectedResponseExceptionType;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.exception.ResourceModifiedException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.serializer.SerializerAdapter;
+import reactor.core.publisher.Mono;
+
+/**
+ * Implementation for calling Non-Azure OpenAI service
+ */
+public final class OpenAIClientNonAzureImpl {
+    /** The proxy service used to perform REST calls. */
+    private final OpenAIClientNonAzureService service;
+
+    /** The HTTP pipeline to send requests through. */
+    private final HttpPipeline httpPipeline;
+
+    /**
+     * Gets The HTTP pipeline to send requests through.
+     *
+     * @return the httpPipeline value.
+     */
+    public HttpPipeline getHttpPipeline() {
+        return this.httpPipeline;
+    }
+
+    /** The serializer to serialize an object into a string. */
+    private final SerializerAdapter serializerAdapter;
+
+    /**
+     * Gets The serializer to serialize an object into a string.
+     *
+     * @return the serializerAdapter value.
+     */
+    public SerializerAdapter getSerializerAdapter() {
+        return this.serializerAdapter;
+    }
+
+    /**
+     * This is the endpoint that non-azure OpenAI supports. Currently, it has only v1 version.
+     */
+    public static final String OPEN_AI_ENDPOINT = "https://api.openai.com/v1";
+
+    /**
+     * Initializes an instance of OpenAIClient client.
+     *
+     * @param httpPipeline The HTTP pipeline to send requests through.
+     * @param serializerAdapter The serializer to serialize an object into a string.
+     */
+    public OpenAIClientNonAzureImpl(HttpPipeline httpPipeline, SerializerAdapter serializerAdapter) {
+        this.httpPipeline = httpPipeline;
+        this.serializerAdapter = serializerAdapter;
+        this.service = RestProxy.create(OpenAIClientNonAzureService.class, this.httpPipeline, this.getSerializerAdapter());
+    }
+
+    /**
+     * The interface defining all the services for OpenAIClient to be used by the proxy service to perform REST calls.
+     */
+    @Host("{endpoint}")
+    @ServiceInterface(name = "OpenAIClient")
+    public interface OpenAIClientNonAzureService {
```

Comment:
Is this interface different from the Azure service proxy interface?

Reply:
Yes. 
Azure OpenAi Host has a specific pattern but Non-Azure OpenAI has a fixed endpoint and Non-Azure OpenAI has no PATH method parameter.
Azure URL
```
 @Host("{endpoint}/openai")
    @ServiceInterface(name = "OpenAIClient")
    public interface OpenAIClientService {
        @Post("/deployments/{deploymentId}/embeddings")
```
NonAzure URL
```
 @Host("{endpoint}")
    @ServiceInterface(name = "OpenAIClient")
    public interface OpenAIClientNonAzureService {
        @Post("/embeddings")
        @ExpectedResponses({200})
```